staging/src/k8s.io/kubectl/pkg/cmd/apply/apply.go

```diff
@@ -172,9 +172,12 @@ func NewCmdApply(baseName string, f cmdutil.Factory, ioStreams genericclioptions
 		Long:                  applyLong,
 		Example:               applyExample,
 		Run: func(cmd *cobra.Command, args []string) {
-			cmdutil.CheckErr(o.Complete(f, cmd))
+			cmdutil.CheckErr(o.RecordFlags.Complete(cmd))
+			af, err := cmdutil.GetApplyFlags(cmd)
+			af.FieldManager = GetApplyFieldManagerFlag(cmd, o.ServerSideApply)
+			cmdutil.CheckErr(err)
+			cmdutil.CheckErr(o.Complete(f, af))
```

Comment:
This is wrong, ideally this should have only 3 invocations:
1. Complete
2. Validate
3. Run.